### PR TITLE
RIIR: Integration tests

### PR DESCRIPTION
**Pull request by flip1995**
In #4825 the `rust-lang/chalk` test failed because the output was too large. I didn't want to completely disabling the output, since showing the backtrace of an ICE directly in travis is pretty useful. Since finding strings in command outputs is easier in Rust, than in bash, I just RIIRed it.

This and also rewriting our tests in Rust may help with trying out new CI platforms (cc #4577)

changelog: none

### CODE REVIEW

**Comment by phansch** [comment]
@bors try
Let's try with a working rustc again :)

**Comment by flip1995** [comment]
This takes about 1 minute longer per integration test. I think this is because of the additional deps and that `libgit2` doesn't support shallow clones (https://github.com/rust-lang/git2-rs/issues/190#issuecomment-280709426). (Maybe that changed by now)

I think this is worth though.

**Comment by phansch** [comment]
re the shallow clones: It seems like there is active work going on at https://github.com/libgit2/libgit2/issues/3058

The 1 minute extra time is fine with me :+1:

File: tests/integration.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    } else if stderr.contains("query stack during panic") {
+        panic!("query stack during panic in the output");
+    } else if stderr.contains("E0463") {
+        panic!("error: E0463");
+    }
+
+    match output.status.code() {
+        Some(code) => {
+            if code == 0 {
+                eprintln!("Compilation successful");
```
Comment (phansch) [inline tests/integration.rs]:
```suggestion
                println!("Compilation successful");
```

**Pull request merged**